### PR TITLE
Enable default schema check and deprecate -s

### PR DESCRIPTION
**Pull request by soumeh01**
## Fixes
- Addressing https://github.com/Open-CMSIS-Pack/devtools/issues/1777

## Changes
- Deprecation of `-s` option
- Enabled default schema check
- Added new option to disable schema check `-n,--no-schema-check`

## Checklist

- [x] 🤖 This change is covered by unit tests as required.
- [x] 🤹 All required manual testing has been performed.
- [x] 🛡️ Security impacts have been considered.
- [x] 📖 All documentation updates are complete.
- [x] 🧠 This change does not change third-party dependencies

### CODE REVIEW

File: cmd/cbuild/commands/build/buildcprj.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func BuildCPRJ(cmd *cobra.Command, args []string) error {
 	debug, _ := cmd.Flags().GetBool("debug")
 	verbose, _ := cmd.Flags().GetBool("verbose")
 	clean, _ := cmd.Flags().GetBool("clean")
-	schema, _ := cmd.Flags().GetBool("schema")
+	schemaChk, _ := cmd.Flags().GetBool("no-schema-check")
```
Comment (jkrech) [inline cmd/cbuild/commands/build/buildcprj.go:46]: IMHO the negation should be in the variable name:
`noSchemaChk`

**Review by jkrech — CHANGES_REQUESTED** [review]
Other than the variable name this looks good to me.

File: cmd/cbuild/commands/build/buildcprj.go (hunk 1 above)
Comment (soumeh01) [reply cmd/cbuild/commands/build/buildcprj.go:46]: We use the hyphen as a separator to ensure consistency with other options that already follow this convention, such as `frozen-packs`, `context-set`, and `update-rte`.

Comment (soumeh01) [reply cmd/cbuild/commands/build/buildcprj.go:46]: Addressed in [Renamed variables](https://github.com/Open-CMSIS-Pack/cbuild/pull/336/commits/ff499f2fca4762bab1399b6ff34966c8f7be0f76)

**Review by jkrech — APPROVED** [review]
LGTM

**Pull request merged**